### ISSUE TITLE
Little pixels with random color (from the theme) sprinkles at the top of the screen while typing

### ISSUE THREAD

**Issue by pingshunhuangalex** [Miodec/monkeytype#568]
I don't have a screenshot for this issue but it's quite easy to reproduce.
- use a custom theme
- make sure timer at the top of the screen is turned on (in the form of a bar)
- pay attention at the top of the screen (progress bar area) and start typing
- little pixels with random colour (from the theme) sprinkles/flashes at the top of the screen while typing

**Comment by Miodec** [comment]
Cant reproduce. Probably just a visual bug? How consistent is it?

**Comment by pingshunhuangalex** [comment]
Yeah just a visual bug and I can stably reproduce it. You need to be patient when typing and need to pay close attention to the progress bar section.

Feel free to use my settings:
```
{
  "theme": "terminal",
  "customTheme": true,
  "customThemeColors": [
    "#212121",
    "#c792ea",
    "#c792ea",
    "#4f6875",
    "#eeffff",
    "#ff5370",
    "#ef5350",
    "#ff5370",
    "#ef5350"
  ],
  "favThemes": ["terminal", "serika_dark"],
  "showKeyTips": false,
  "showLiveWpm": false,
  "showTimerProgress": true,
  "smoothCaret": true,
  "quickTab": true,
  "punctuation": false,
  "numbers": false,
  "words": 50,
  "time": 60,
  "mode": "time",
  "quoteLength": 0,
  "language": "english_10k",
  "fontSize": "15",
  "freedomMode": true,
  "resultFilters": {
    "words": {
      "10": false,
      "25": false,
      "50": false,
      "100": false,
      "200": false,
      "custom": false
    },
    "tags": { "lJwVmUz448oVdbb77rnc": true, "none": false },
    "difficulty": { "master": false, "expert": false, "normal": true },
    "mode": { "words": false, "quote": false, "time": true, "custom": false },
    "punctuation": { "off": true, "on": false },
    "time": {
      "15": false,
      "30": false,
      "60": true,
      "120": false,
      "custom": false
    },
    "language": {
      "hungarian": false,
      "thai": false,
      "english_1k": false,
      "slovak": false,
      "italian": false,
      "indonesian": false,
      "english_expanded": false,
      "english": false,
      "portuguese": false,
      "english_10k": true,
      "german": false,
      "spanish": false,
      "french": false,
      "norwegian": false,
      "czech": false,
      "turkish": false,
      "dutch": false,
      "hebrew": false,
      "russian": false,
      "polish": false
    },
    "numbers": { "off": true, "on": false },
    "funbox": {
      "58008": false,
      "none": true,
      "earthquake": false,
      "mirror": false,
      "tts": false,
      "ascii": false,
      "choo_choo": false,
      "layoutfluid": false,
      "simon_says": false,
      "rAnDoMcAsE": false,
      "plus_one": false,
      "space_balls": false,
      "nausea": false,
      "gibberish": false,
      "specials": false,
      "read_ahead": false,
      "round_round_baby": false
    },
    "date": {
      "last_month": false,
      "all": true,
      "last_week": false,
      "last_day": false
    }
  },
  "difficulty": "normal",
  "blindMode": false,
  "quickEnd": true,
  "caretStyle": "default",
  "paceCaretStyle": "block",
  "flipTestColors": false,
  "capsLockBackspace": false,
  "layout": "default",
  "savedLayout": "default",
  "confidenceMode": "off",
  "indicateTypos": true,
  "timerStyle": "bar",
  "colorfulMode": true,
  "randomTheme": "off",
  "timerColor": "main",
  "timerOpacity": "0.25",
  "stopOnError": "off",
  "showAllLines": false,
  "keymapMode": "off",
  "keymapStyle": "staggered",
  "keymapLayout": "colemak_dh",
  "fontFamily": "Roboto_Mono",
  "smoothLineScroll": true,
  "alwaysShowDecimalPlaces": false,
  "alwaysShowWordsHistory": false,
  "singleListCommandLine": "manual",
  "playSoundOnError": true,
  "playSoundOnClick": "off",
  "startGraphsAtZero": false,
  "swapEscAndTab": true,
  "showOutOfFocusWarning": true,
  "paceCaret": "pb",
  "paceCaretCustomSpeed": 100,
  "pageWidth": "100",
  "chartAccuracy": true,
  "chartStyle": "scatter",
  "minWpm": "off",
  "minWpmCustomSpeed": 100,
  "highlightMode": "letter",
  "alwaysShowCPM": false,
  "enableAds": "off",
  "hideExtraLetters": false,
  "strictSpace": true
}
```

**Comment by TheeKevin** [comment]
@Miodec it's been quite some time since this issue was created so I'm wondering if this should be closed since @pingshunhuangalex is the only one who's encountered this problem which might mean this bug doesn't exist anymore? Plus he says that he can stably reproduce it but can't provide a screenshot which could be suspicious but could also mean that he doesn't encounter this bug anymore.

**Comment by pingshunhuangalex** [comment]
Just tried it and I can still see those random pixel dots on the timer progress bar.

Here's the video. I added a slow-mo when those dots showed up so you won't miss it. It takes some designer's eye to capture those. Not something everybody would notice or care about, but definitely not `suspicious` cuz @TheeKevin bank won't send me any money if I report bugs like this on Github...

https://user-images.githubusercontent.com/17170199/110873416-d1186e80-8325-11eb-9fac-2cb56aa31066.mp4

Whether the ticket should be closed depends on if the issue has been resolved or if the team has the capacity to fix it, but definitely not `Oh, it's been there for a while, and I can't reproduce it. Let's close it`. Having said that, I don't mind the ticket is closed knowing it won't be addressed any time soon or not at all.

**Comment by Miodec** [comment]
Well, i still have not encountered it and I've been trying to use the bar timer as much as possible to try to see it for myself.

In the title you mentioned many small ones but in the video I can see one big one. Is it random?
Are they always in the same spot?
Are they visible for long?
Their color - is it consistent with other colors of the currently active theme, or different?

**Comment by TheeKevin** [comment]
@pingshunhuangalex sorry I just figured that this issue might have been sorted on discord (since I don't use discord) plus since nothings happened for quite some time I also figured that the bug might've simply disappeared.

**Comment by pingshunhuangalex** [comment]
> In the title you mentioned many small ones but in the video I can see one big one. Is it random?
In the video, there were 2 dots/blocks showing up in slow-mo. This is just a short time window I cropped to create some focus. If I keep typing, the dots will keep showing up.

The fact that you only saw one and missed the other (in a slow-mo) proved how easily they can be overlooked. They are more or less similar sizes, and I'd call them small. The size doesn't seem random to me, but the place of occurrence is.

> Are they always in the same spot?
No, they show up in different places or at least multiple places that make them look random enough. However, they only appear near/on the progress bar area.

> Are they visible for long?
They only occur for a fraction of a second. Combining with the random position, that's why they look like sprinkles/flashes to me.

> Their color - is it consistent with other colors of the currently active theme, or different?
The video showed cyan so I'd say no, at least not a direct pick of any of the theme color I put in, but it could be the opposite color of the theme ones. There's no way I can tell if there are any color transformation going on behind the curtain

**Comment by Miodec** [comment]
> The video showed cyan so I'd say no, at least not a direct pick of any of the theme color I put in, but it could be the opposite color of the theme ones. There's no way I can tell if there are any color transformation going on behind the curtain

I think this is the main thing we should try to determine first - try to use different themes to find out if its using a theme color or not.

If it is, it could be an actual website element. If it's not, then I don't know.

**Comment by pingshunhuangalex** [comment]
Confirmed fixed with the big rewrite in v1.8
